added bill pagination

JonathanTylerWeber opened the pull request:
**Issue**
-Paginate Session Bills

**Solution**
-Manually paginate data from getMasterList and implement it using search params

**Implementation Details**
-I created a getPage function to sort through the data by pages. On the Bills page I get the current page from the search params and set the page length as well as getting total pages. Using the getPage function I create a billList for the currentPage. I created a component PaginationControls as well as PaginationControlsProps for the types. Pagination Controls has the next and prev buttons which are hidden if they can't go further. It displays the currentPage and totalPages passed down through the bills page.

**Testing**
-You can go to a Session Bills page and scroll to the bottom. There are 48 per page to fit our width of bills and on the first page the prev button will be hidden. After clicking Next it will show you the next page of bills. On the last page the next button will be hidden.

**Additional Information**
-There is a getSearch method for session bills that has pagination but it doesn't include session data and the session data should be shown at the top of the session bills page. I don't know if it's possible to send the session data down through the link from the sessions page. So on the session bills page rather than using both getSearch for the bill pagination and getMasterList just for data on a single session, I paginated the session bills manually.

SevanBadal commented on app/state/[state]/session/[session]/page.tsx:
```diff
@@ -1,13 +1,20 @@
     try {
-        const legiscanApiKey = process.env.LEGI_KEY; 
+        const legiscanApiKey = process.env.LEGI_KEY;
         const res = await fetch(`https://api.legiscan.com/?key=${legiscanApiKey}&op=getMasterList&id=${sessionID}`);
```
`getSearch` does support paging so I think we should implement that here instead of using `getMasterList`. To get around the lack of session data from `getSearch`, we can implement a second async function to fetch session data by utilizing the `getSessionPeople` legibill operation that returns both the session object and "people" involved in the session (we can ignore the people array for now but might be cool to have a /session/[session]/people page in the future!). 

for ex session/[session]/page.tsx will has two separate functions to handle bills and session data: 
```typescript
export default async function Page({ params, searchParams }: { params: { session: number; state: string }, searchParams: { page: string } }) {
    // getSessionBills uses legibills getSearch endpoint and leverages their paging support
    const billList: someType = await getSessionBills(params.session, otherParamsLikePageAndPageSize)
    // getSessionData uses getSessionPeople which will give you back an object that contains the session data
    const sessionData: someType = await getSessionData(params.session)

    // other stuffz
}
```

It would be good to model the types correctly here. After you implement these changes, I'll come in and clean up the types before merging.

SevanBadal replied:
Also, you may need to call `getSessionsBill` from a child component or create a layout for this page and call `getSessionData` from that to prevent fetching session data every time a page changes @JonathanTylerWeber

SevanBadal reviewed (changes requested):
Review my comments! Let's lean into the paging params supported by `getSearch` and use `getSessionPeople` to fetch the session object separately!

JonathanTylerWeber commented:
-In order to separate getting the session data and bills list I created a layout.tsx for the Session Bills page. The layout uses getSessionPeople in order to get the session data as we discussed to avoid fetching session data for each page of bills.

-On the Session Bills page I updated it to use getSearch for pagination. The bill data returned from getSearch is different (keys have different names and description and status date aren't available).

-I made a new transform data function and updated the apiResponse data type to work with getSearch 

-I added new data types to support this change: getSearchBill(returns different keys), sessionPeopleResponse, sessionSummary, and people

JonathanTylerWeber commented:
Another small thing but I accidentally named some new data types not following the convention set starting with lowercase. I tried fixing it but it seems like that might not have worked in that commit.

SevanBadal reviewed (approved):
Looks good! Thanks for implementing those changes. I left a few comments for your review. Merging!

JonathanTylerWeber commented on app/state/[state]/session/[session]/page.tsx:
```diff
@@ -1,56 +1,57 @@
+                            <p className='line-clamp-1'><strong>Bill Number:</strong> {bill.bill_number}</p>
+                            <p className='line-clamp-1'><strong>Last Action:</strong> {bill.last_action}</p>
+                            <p className='line-clamp-1'><strong>Last Action Date:</strong> {bill.last_action_date}</p>
+                            {/* <p className='line-clamp-1'><strong>Status Date:</strong> {bill.status_date}</p> */}
```
got it
